src/pages/projects/_components/ProjectList.tsx

```diff
@@ -1,55 +1,85 @@
-import { faEdit, faEnvelope } from "@fortawesome/free-regular-svg-icons";
-import { faTrash } from "@fortawesome/free-solid-svg-icons";
-import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { mdiDelete, mdiSquareEditOutline } from "@mdi/js";
+import Icon from "@mdi/react";
+import { useEffect, useRef, useState } from "react";
 import { Link } from "react-router-dom";
-import { ProjectListresponse, ProjectResponse } from "../../../features/project";
+import Modal from "../../../components/Modal";
+import { ProjectResponse } from "../../../features/project";
 
 type Props = {
-    projectList: ProjectResponse[], 
+    projectList: ProjectResponse[],
     deleteProject: (id: string) => void
 };
 
-const ProjectList = ({projectList, deleteProject}: Props) => {
+const ProjectList = ({ projectList, deleteProject }: Props) => {
     const projects = projectList;
-    
-    
-    
-    return (  
-        <table >
+
+    const [dialog, setDialog] = useState({
+        message: "",
+        openModal: false,
+        item: "",
+        title: ""
+    });
+
+    let projectId = "";
```

Comment:
This is going to get reinitialize on every component paint, use a  useState hook instead

Reply:
fixed